src/Core/Authentication.php

```diff
@@ -0,0 +1,213 @@
+		$_SESSION['authenticated'] = 1;
+		$_SESSION['page_flags'] = $user_record['page-flags'];
+		$_SESSION['my_url'] = $a->getbaseUrl() . '/profile/' . $user_record['nickname'];
+		$_SESSION['my_address'] = $user_record['nickname'] . '@' . substr($a->getbaseUrl(), strpos($a->getbaseUrl(), '://') + 3);
+		$_SESSION['addr'] = defaults($_SERVER, 'REMOTE_ADDR', '0.0.0.0');
+
+		$a->user = $user_record;
+
+		if ($interactive) {
+			if ($a->user['login_date'] <= NULL_DATE) {
+				$_SESSION['return_url'] = 'profile_photo/new';
+				$a->module = 'profile_photo';
+				info(L10n::t("Welcome ") . $a->user['username'] . EOL);
+				info(L10n::t('Please upload a profile photo.') . EOL);
+			} else {
+				info(L10n::t("Welcome back ") . $a->user['username'] . EOL);
+			}
+		}
+
+		$member_since = strtotime($a->user['register_date']);
+		if (time() < ($member_since + ( 60 * 60 * 24 * 14))) {
+			$_SESSION['new_member'] = true;
+		} else {
+			$_SESSION['new_member'] = false;
+		}
+		if (strlen($a->user['timezone'])) {
+			date_default_timezone_set($a->user['timezone']);
+			$a->timezone = $a->user['timezone'];
+		}
+
+		$master_record = $a->user;
+
+		if ((x($_SESSION, 'submanage')) && intval($_SESSION['submanage'])) {
+			$user = DBA::selectFirst('user', [], ['uid' => $_SESSION['submanage']]);
+			if (DBA::isResult($user)) {
+				$master_record = $user;
+			}
+		}
+
+		if ($master_record['parent-uid'] == 0) {
+			// First add our own entry
+			$a->identities = [['uid' => $master_record['uid'],
+					'username' => $master_record['username'],
+					'nickname' => $master_record['nickname']]];
+
+			// Then add all the children
+			$r = DBA::select('user', ['uid', 'username', 'nickname'],
+				['parent-uid' => $master_record['uid'], 'account_removed' => false]);
+			if (DBA::isResult($r)) {
+				$a->identities = array_merge($a->identities, DBA::toArray($r));
+			}
+		} else {
+			// Just ensure that the array is always defined
+			$a->identities = [];
+
+			// First entry is our parent
+			$r = DBA::select('user', ['uid', 'username', 'nickname'],
+				['uid' => $master_record['parent-uid'], 'account_removed' => false]);
+			if (DBA::isResult($r)) {
+				$a->identities = DBA::toArray($r);
+			}
+
+			// Then add all siblings
+			$r = DBA::select('user', ['uid', 'username', 'nickname'],
+				['parent-uid' => $master_record['parent-uid'], 'account_removed' => false]);
+			if (DBA::isResult($r)) {
+				$a->identities = array_merge($a->identities, DBA::toArray($r));
+			}
+		}
+
+		$r = DBA::p("SELECT `user`.`uid`, `user`.`username`, `user`.`nickname`
+			FROM `manage`
+			INNER JOIN `user` ON `manage`.`mid` = `user`.`uid`
+			WHERE `user`.`account_removed` = 0 AND `manage`.`uid` = ?",
+			$master_record['uid']
+		);
+		if (DBA::isResult($r)) {
+			$a->identities = array_merge($a->identities, DBA::toArray($r));
+		}
+
+		if ($login_initial) {
+			logger('auth_identities: ' . print_r($a->identities, true), LOGGER_DEBUG);
```

Comment:
Watch out for global functions, their corresponding file should be `require_once` by this file until we move them to classes.